packages/react/src/menubar/root/useMenubarRoot.ts

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import {
+  FloatingRootContext,
+  useFloatingRootContext,
+  useInteractions,
+  useListNavigation,
+} from '@floating-ui/react';
+import { type MenuOrientation } from '../../menu/root/useMenuRoot';
+import { GenericHTMLProps } from '../../utils/types';
+import { useDirection } from '../../direction-provider/DirectionContext';
+
+export function useMenubarRoot(parameters: useMenubarRoot.Parameters): useMenubarRoot.ReturnValue {
+  const { orientation, disabled, loop } = parameters;
+
+  const [contentElement, setContentElement] = React.useState<HTMLElement | null>(null);
+  const [activeIndex, setActiveIndex] = React.useState<number | null>(0);
+  const [hasSubmenuOpen, setHasSubmenuOpen] = React.useState(false);
+  const direction = useDirection();
+
+  const floatingRootContext = useFloatingRootContext({
+    elements: {
+      reference: null,
+      floating: contentElement,
+    },
+    open: true,
+  });
+
+  const itemDomElements = React.useRef<(HTMLElement | null)[]>([]);
+  const itemLabels = React.useRef<(string | null)[]>([]);
+
+  const listNavigation = useListNavigation(floatingRootContext, {
+    enabled: !disabled,
+    listRef: itemDomElements,
+    activeIndex,
+    nested: false,
+    loop,
+    orientation,
+    rtl: direction === 'rtl',
+    onNavigate: (index) => {
+      if (index !== null) {
+        setActiveIndex(index);
+      }
+    },
+  });
```

Comment:
I notice this has a lot of bugs, wouldn't it be better just to use `Composite`? This is missing the hover functionality if a menu is already open, but it avoids many issues by default

https://github.com/floating-ui/floating-ui/blob/master/packages/react/test/visual/components/Menubar.tsx

Reply:
I'll see if it's any better, thanks.